packages/framework/presence/src/systemWorkspace.ts

```diff
@@ -0,0 +1,192 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/core-utils/internal";
+
+import type { ClientConnectionId } from "./baseTypes.js";
+import type { InternalTypes } from "./exposedInternalTypes.js";
+import type { ClientSessionId } from "./presence.js";
+import type { PresenceStatesInternal } from "./presenceStates.js";
+import type { PresenceStates, PresenceStatesSchema } from "./types.js";
+
+import {
+	createEmitter,
+	type ISubscribable,
+} from "@fluid-experimental/presence/internal/events";
+
+/**
+ * @internal
+ */
+export interface SystemWorkspaceDatastore {
+	clientToSessionId: {
+		[ConnectionId: ClientConnectionId]: InternalTypes.ValueRequiredState<ClientSessionId>;
+	};
+}
+
+interface SystemWorkspace {}
+
+class SystemWorkspaceImpl implements PresenceStatesInternal, SystemWorkspace {
+	public constructor(private readonly datastore: SystemWorkspaceDatastore) {}
+
+	public ensureContent<TSchemaAdditional extends PresenceStatesSchema>(
+		_content: TSchemaAdditional,
+	): never {
+		throw new Error("Method not implemented.");
+	}
+
+	public processUpdate(
+		_received: number,
+		_timeModifier: number,
+		remoteDatastore: {
+			clientToSessionId: {
+				[
+					ConnectionId: ClientConnectionId
+				]: InternalTypes.ValueRequiredState<ClientSessionId> & {
+					ignoreUnmonitored?: true;
+				};
+			};
+		},
+	): void {
+		for (const [clientConnectionId, value] of Object.entries(
+			remoteDatastore.clientToSessionId,
+		)) {
+			const knownSessionId: InternalTypes.ValueRequiredState<ClientSessionId> | undefined =
+				this.datastore.clientToSessionId[clientConnectionId];
+			if (knownSessionId === undefined) {
+				this.datastore.clientToSessionId[clientConnectionId] = value;
+			} else {
+				assert(knownSessionId.value === value.value, "Mismatched SessionId");
+			}
+			value: this.selfAttendee.sessionId,
+		};
+
+		this.selfAttendee.currentConnectionId = () => clientConnectionId;
+		this.attendees.set(clientConnectionId, this.selfAttendee);
+	}
+
+	public getAttendees(): ReadonlySet<ISessionClient> {
+		return new Set(this.attendees.values());
+	}
+
+	public getAttendee(clientId: ClientConnectionId | ClientSessionId): ISessionClient {
```

Comment:
Could you describe scenarios when you would use client connection id vs client session id?

Reply:
The system today offers plenty of client connection ids (just called clientId). Presence feature adds ClientSessionId and makes them available. You use one or the other based on what you have. For example, current service audience will provide a list of client connection ids for a user. You would use client connection id to get the session client.